packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/document/services/DocumentSanitizedJsonContentProvider.java

```diff
@@ -120,6 +128,23 @@ public Optional<String> getContent(ResourceSet resourceSet, String name, InputSt
         return optionalContent;
     }
 
+    /**
+     * Give fresh, unique IDs to all the elements in the resource.
+     */
```

Comment:
Improvements have been made to the performance of the upload among other things, can't we do better than an additional eAllContents? Why is that necessary to iterate once again on everything?

Reply:
This is more of an "emergency fix", and I did not want to make more signigicant changes for the moment.

My guess is that somehting like this could actually completely replace DocumentSanitizedJsonContentProvider (whose whole point is to ensure the imported elements has fresh, non-conflicting ids), but that would invole more stuctural changes.